python/setup.py

```diff
@@ -23,21 +26,7 @@
     download_url="https://github.com/src-d/hercules",
     packages=["labours"],
     keywords=["git", "mloncode", "mining software repositories", "hercules"],
-    install_requires=[
-        "clint>=0.5.1,<1.0",
-        "matplotlib>=2.0,<4.0",
-        "numpy>=1.12.0,<2.0",
-        "pandas>=0.20.0,<1.0",
-        "PyYAML>=3.0,<5.0",
-        "scipy>=0.19.0,<1.2.2",
-        "protobuf>=3.5.0,<4.0",
-        "munch>=2.0,<3.0",
-        "hdbscan>=0.8.0,<2.0",
-        "seriate>=1.0,<2.0",
-        "fastdtw>=0.3.2,<2.0",
-        "python-dateutil>=2.6.0,<3.0",
-        "lifelines>=0.20.0,<2.0",
-    ],
+    install_requires=requirements,
```

Comment:
The old `requirements.txt` came from the time I did not care much about reproducibility... Can you please move the interval-ed version constraints here in `setup.py` and generate a proper `requirements.txt` file with `==` version constraints.

Reply:
Sure thing. `pip-tools` to the rescue!